### PR TITLE
Adding map matching section to Java SDK area

### LINKED ISSUE #541: Add section on Java SDK's Map Matching service
[Java SDK section of our docs](https://www.mapbox.com/android-docs/java-sdk/overview/) needs a page for the Java SDK's MapMatching service https://github.com/mapbox/mapbox-java/blob/master/services-matching/src/main/java/com/mapbox/api/matching/v5/MapboxMapMatching.java

### PR DESCRIPTION
**Pull request by langsmith**
Resolves #541 by adding a section about map matching

cc @osana in case you think there's anything else that should be added.

screencapture-localhost-8080-android-docs-java-overview-map-matching-2018-07-20-16_22_51

### CODE REVIEW

**Comment by osana** [comment]
@langsmith @danesfeder  The only thing I could think of is MapMatchingMatching to DirectionRoute functionality.

**Comment by danesfeder** [comment]
@osana @langsmith this is MapMatching to DirectionsRoute functionality is highlighted in navigation currently > https://www.mapbox.com/android-docs/navigation/overview/map-matching/

Maybe we can reference these docs in the docs being added in this PR?

**Comment by langsmith** [comment]
>Maybe we can reference these docs in the docs being added in this PR?

I link to  https://www.mapbox.com/android-docs/navigation/overview/map-matching/ in the `Navigation` section of the docs being added in this pr. See the bottom of the OP screenshot.

File: src/pages/java/overview/map-matching.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  - title: "Navigation"
+
+---
+
+The Mapbox Map Matching API snaps fuzzy, inaccurate traces from a GPS unit or a device (such as a phone) to the OpenStreetMap road and path network using the Mapbox Directions API. The result is a clean path that can be displayed on a map or used for other analysis. The API can receive a coordinate list with a size of 2 to 100.  Make sure to [have a look at the API documentation if you want more information](https://www.mapbox.com/api-documentation/#map-matching).
+
+
+## API request
+
+To begin with, you'll need to create a new instance of the `MapboxMapMatching` object and use its builder to customize your matching request. The options offered in the builder include choices such as whether to return steps and turn-by-turn instructions or the format of the returned geometry.
```
Comment (colleenmcginnis) [inline src/pages/java/overview/map-matching.md]: >To begin ~~with~~

Comment (colleenmcginnis) [reply src/pages/java/overview/map-matching.md]: >The options offered in the builder include choices such as whether to return steps and turn-by-turn instructions or the format of the returned geometry.

to 

>The options offered in the builder include whether to return steps and turn-by-turn instructions and the format of the returned geometry.

File: src/pages/java/overview/map-matching.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    .accessToken(Mapbox.getAccessToken())
+    .profile(PROFILE_DRIVING)
+    .coordinates(listOfPoints)
+    .annotations(ANNOTATION_SPEED)
+    .overview(OVERVIEW_FULL)
+    .steps(false)
+    .build();
+```
+
+Make sure to use the `DirectionsCriteria` to reference the default Mapbox directions profile styles such as driving or walking.
```
Comment (colleenmcginnis) [inline src/pages/java/overview/map-matching.md]: >Make sure to use the `DirectionsCriteria` to reference the default Mapbox directions profile styles such as driving or walking.

to 

>Make sure to use the `DirectionsCriteria` to reference the default Mapbox directions profiles such as driving or walking.

File: src/pages/java/overview/map-matching.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+
+## API response
+
+After creating the `MapboxMapMatching` instance with all your customization parameters, you'll need to handle the API response. Like all API calls inside of the Mapbox Java SDK, the response will come inside a Retrofit callback. Inside `onResponse()`, you can access the API's returned response if successful.
```
Comment (colleenmcginnis) [inline src/pages/java/overview/map-matching.md]: >inside of

to 

>made with

?

File: src/pages/java/overview/map-matching.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  @Override
+  public void onFailure(Call<MapMatchingResponse> call, Throwable throwable) {
+    
+  }
+});
+```
+
+##  Navigation
+
+Visit [https://www.mapbox.com/android-docs/navigation/overview/map-matching/](https://www.mapbox.com/android-docs/navigation/overview/map-matching/) for more information about how to use the Map Matching API with [the Mapbox Navigation SDK for Android](https://www.mapbox.com/android-docs/navigation/overview/).
```
Comment (colleenmcginnis) [inline src/pages/java/overview/map-matching.md]:
```
Visit the [Map matching guide](https://www.mapbox.com/android-docs/navigation/overview/map-matching/) ... 
```

File: src/pages/java/overview/map-matching.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+title: "Map matching"
+description: "Official documentation on the Mapbox Java SDK Map Matching API"
+sideNavSections:
+  - title: "API request"
+  - title: "API response"
+  - title: "Navigation"
+
+---
+
+The Mapbox Map Matching API snaps fuzzy, inaccurate traces from a GPS unit or a device (such as a phone) to the OpenStreetMap road and path network using the Mapbox Directions API. The result is a clean path that can be displayed on a map or used for other analysis. The API can receive a coordinate list with a size of 2 to 100.  Make sure to [have a look at the API documentation if you want more information](https://www.mapbox.com/api-documentation/#map-matching).
```
[the remark this answers is not in the document]
Comment (langsmith) [reply src/pages/java/overview/map-matching.md]: Fixed

File: src/pages/java/overview/map-matching.md (hunk 1 above)
Comment (langsmith) [reply src/pages/java/overview/map-matching.md]: Fixed

File: src/pages/java/overview/map-matching.md (hunk 2 above)
Comment (langsmith) [reply src/pages/java/overview/map-matching.md]: Fixed

File: src/pages/java/overview/map-matching.md (hunk 3 above)
Comment (langsmith) [reply src/pages/java/overview/map-matching.md]: Fixed

File: src/pages/java/overview/map-matching.md (hunk 4 above)
Comment (langsmith) [reply src/pages/java/overview/map-matching.md]: Fixed

**Pull request merged**